### PR TITLE
Feature/id and refs

### PR DESCRIPTION
**Pull request by steckhelena**
#### What problem is this solving?

This feature branch implements the json schema [$ref](https://tools.ietf.org/html/draft-wright-json-schema-01#section-8) and [$id](https://tools.ietf.org/html/draft-wright-json-schema-01#section-9.2) resolving, as well as changes the preffix for accessing the instantiated schema members to not be the same as a [URI Fragment Identifier Representation of a JSON pointer](https://tools.ietf.org/html/rfc6901#section-6).

#### Checklist/Reminders

- [x] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Clubhouse story (if applicable).
- [x] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
✔️ | Bug fix 
✔️ | New feature 
✔️ | Breaking change 
✔️ | Technical improvements

### CODE REVIEW

File: src/JSONSchema/logic/refHandlers.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
+import { JSONSchemaType, IDSchemaPair } from '../types'
+import { getSplitPointer, concatFormPath } from './pathUtils'
+
+const isAbsoluteURI = (uri: string) => {
+  const absoluteRegExp = new RegExp(/^[a-z][a-z0-9+.-]*:/, 'i')
```
Comment (kaisermann) [inline src/JSONSchema/logic/refHandlers.ts]: You can make this `RegExp` a top-level variable to prevent it being creating over and over again

Comment (kaisermann) [reply src/JSONSchema/logic/refHandlers.ts]: Also: 
```suggestion
  const absoluteRegExp = /^[a-z][a-z0-9+.-]*:/i
```

File: src/JSONSchema/logic/refHandlers.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  return absoluteRegExp.test(uri)
+}
+
+const isURIFragmentPointer = (pointer: string) => {
+  const fragmentRegExp = new RegExp(/^#(\/(([^#/~])|(~[01]))*)*/, 'i')
```
Comment (kaisermann) [inline src/JSONSchema/logic/refHandlers.ts]: Same thing as commented above

File: src/hooks/validators/getGenericValidator.ts [hunk 3]
```diff
@@ -65,16 +66,16 @@ export const getValidator = (
       : undefined),
   }
 
-  if (required) {
+  if (isRequired) {
```
Comment (kaisermann) [inline src/hooks/validators/getGenericValidator.ts]: 💅

File: src/JSONSchema/logic/pathUtils.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  if (split[0] === '#') {
+    split.shift()
+  }
+  // If there was a '/' at the end of the path before it is split(it should not
+  // be there) this removes the non-existent path
+  if (split[split.length - 1] === '') {
+    split.pop()
+  }
+  return split
+}
```
[the remark this answers is not in the document]
Comment (steckhelena) [reply src/JSONSchema/logic/pathUtils.ts]: xD

**Pull request merged**